samples/packages/awscdk-typescript/src/package-config.ts

```diff
@@ -183,20 +188,21 @@ export class PackageComponentStorageBucketConfig implements PackageComponentConf
 }
 
 export class PackageComponentStaticWebsiteHostingConfig implements PackageComponentConfig {
+  static TYPING = (manifest as PackageManifest).constructs.components.types[ComponentTyping.DISTRIBUTION_WEBSITE];
+
   constructor(
     public readonly name: string,
     public readonly hostName: string,
     public readonly domain: string, //TODO: figure out how best to handle none AWS hosted DNS management
-  ) { }
+  ) {}
 
   public static has(config: cdf.TraitConfig<PackageComponentConfig>): boolean {
-    return config.type == 'distribution' && config.subtype == 'staticWebsite';
+    return config.type == this.TYPING.type && config.subtype == this.TYPING.subtype;
```

Comment:
In particular for this component type (static website hosting): *I've changed the subtype to `website` as the type `distribution` should have pre-qualified the static nature already*